subsys/bluetooth/host/id.c

```diff
@@ -1266,6 +1262,32 @@ int bt_id_delete(uint8_t id)
 	return 0;
 }
 
+
+int bt_id_delete(uint8_t id)
+{
+	if (id == BT_ID_DEFAULT || id >= bt_dev.id_count) {
+		return -EINVAL;
+	}
+
+	return id_delete(id);
+}
+
+int bt_id_delete_all(void)
+{
+	for (uint8_t id = 0U; id < bt_dev.id_count; id++) {
+		int err;
+
+		err = id_delete(id);
+		if (err) {
+			return err;
+		}
+	}
+
+	bt_dev.id_count = 0;
+
+	return 0;
+}
```

Comment:
This will also delete the `BT_ID_DEFAULT`, which isn't possible via the `bt_id_delete` API. Is this intended? Is the `BT_ID_DEFAULT` added again on `bt_enable`?

Reply:
Yes, `bt_id_init` sets up `BT_ID_DEFAULT` which is called in `bt_enable`